types/accesscontrol/access_operation_map.go

```diff
@@ -0,0 +1,23 @@
+package accesscontrol
+
+// Alias for Map of MessageIndex -> AccessOperation -> Channel
+type MessageAccessOpsChannelMapping = map[int]AccessOpsChannelMapping
+
+// Alias for Map of AccessOperation -> Channel
+type AccessOpsChannelMapping = map[*AccessOperation][]chan interface{}
+
+func WaitForSignals(accessOpsToChannelsMap AccessOpsChannelMapping) {
+	for _, channels := range accessOpsToChannelsMap {
+		for _, channel := range channels {
+			<-channel
+		}
+	}
+}
+
+func SendSignals(accessOpsToChannelsMap AccessOpsChannelMapping) {
+	for accessOp, channels := range accessOpsToChannelsMap {
+		for _, channel := range channels {
+			channel <- accessOp
```

Comment:
I believe we can just write an empty struct to the chann for more efficiency struct{}{}

Reply:
ah good point, updated